framework/uicomponents/qml/Muse/UiComponents/menuitem.cpp

```
@@ -221,77 +265,159 @@ void MenuItem::setSubitems(const QList<MenuItem*>& subitems)
+    return m_subitems;
+}
+
+bool MenuItem::isValid() const
+{
+    return !m_id.isEmpty();
+}
 
-    m_action = action;
-    emit actionChanged();
+// command support
+MenuItem::MenuItem(const rcommand::CommandInfo& info, QObject* parent)
```

Clang (14.0.6)

[warning] constructor does not initialize these fields: QML_ELEMENT, , m_intent, m_shortcuts, m_subitems

(cppcoreguidelines-pro-type-member-init)